encryption_jwt/encryptor.py

```diff
@@ -0,0 +1,83 @@
+import base64
+import logging
+import os
+
+from botocore.exceptions import ClientError
+from cryptography.hazmat.primitives.ciphers.aead import AESGCM
+from temporalio import workflow
+
+with workflow.unsafe.imports_passed_through():
+    import aioboto3
+
+
+class KMSEncryptor:
+    """Encrypts and decrypts using keys from AWS KMS."""
+
+    def __init__(self, namespace: str):
+        self._namespace = namespace
+        self._boto_session = None
+
+    @property
+    def boto_session(self):
+        """Get a KMS client from boto3."""
+        if not self._boto_session:
+            session = aioboto3.Session()
+            self._boto_session = session
+
+        return self._boto_session
+
+    async def encrypt(self, data: bytes) -> tuple[bytes, bytes]:
+        """Encrypt data using a key from KMS."""
+        # The keys are rotated automatically by KMS, so fetch a new key to encrypt the data.
+        data_key_encrypted, data_key_plaintext = await self.__create_data_key(
+            self._namespace
+        )
+
+        if data_key_encrypted is None:
+            raise ValueError("No data key!")
+
+        nonce = os.urandom(12)
+        encryptor = AESGCM(data_key_plaintext)
+        return nonce + encryptor.encrypt(nonce, data, None), base64.b64encode(
```

Comment:
```suggestion
        encrypted = asyncio.get_running_loop().run_in_executor(None, encryptor.encrypt, nonce, data, None)
        return nonce + encrypted, base64.b64encode(
```

I am a little worried about using this under load with CPU bound encryption (and decryption below) blocking the event loop. Technically what is here is ok, but this may make it better, thoughts? Or at least a comment above this encrypt line and the decrypt equivalent below mentioning it? (technically could have put the entire boto3 thread-blocking call under a new thread with this encrypt too if you wanted, sorry I didn't mention it before instead of using aioboto3)